mapie/multi_label_classification.py

```diff
@@ -0,0 +1,746 @@
+            in self.valid_methods_
+        """
+        if self.method not in self.valid_methods_:
+            raise ValueError(
+                "Invalid method. "
+                "Allowed values are" + " or ".join(self.valid_methods_)
+            )
+
+    def _check_all_labelled(self, y: NDArray) -> None:
+        """
+        Check that all observations have at least
+        one label
+
+        Parameters
+        ----------
+        y : NDArray of shape (n_samples, n_labels)
+            Labels of the observations.
+
+        Raises
+        ------
+        ValueError
+            Raise error if at least one observation
+            has no label.
+        """
+        if not (y.sum(axis=1) > 0).all():
+            raise ValueError(
+                "Invalid y. "
+                "All observations should contain at "
+                "least one label."
+            )
+
+    def _check_delta(self, delta: Optional[float]):
+        """
+        Check that delta is not ``None`` when the
+        method is RCPS and that it is between 0 and 1.
+
+        Parameters
+        ----------
+        delta : float
+            Probability with which we control the risk. The higher
+            the probability, the more conservative the algorithm will be.
+
+        Raises
+        ------
+        ValueError
+            If delta is ``None`` and method is RCSP
+        ValueError
+            If delta is not in [0, 1] and method
+            is RCPS.
+        """
+        if (self.method == "rcps") and (delta is None):
+            raise ValueError(
+                "Invalid delta. "
+                "delta cannot be None when using "
+                "RCPS method."
+            )
+        if (not isinstance(delta, float)) and (delta is not None):
+            raise ValueError(
+                "Invalid delta. "
+                f"delta must be a float, not a {type(delta)}"
+            )
+        if (delta is not None) and (
+            ((delta <= 0) or (delta >= 1)) and
+            (self.method == "rcps")
+        ):
+            raise ValueError(
+                "Invalid delta. "
+                "delta must be in ]0, 1["
+            )
+
+    def _check_estimator(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        estimator: Optional[ClassifierMixin] = None,
+    ) -> ClassifierMixin:
+        """
+        Check if estimator is ``None``,
+        and returns a multi-output ``LogisticRegression``
+        instance if necessary.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : ArrayLike of shape (n_samples,)
+            Training labels.
+
+        estimator : Optional[ClassifierMixin], optional
+            Estimator to check, by default ``None``
+
+        Returns
+        -------
+        ClassifierMixin
+            The estimator itself or a default multi-output
+            ``LogisticRegression`` instance.
```

Comment:
What do you mean by that LogisticRegression instance?

Reply:
LogisticRegression is an instance used to run a logistic regression model, in other words, with scikit-learn, when you run a logistic regression, the model type (so the model instance) is LogisticRegression